syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.iotservice";
option java_outer_classname = "IoTServiceProto";
option objc_class_prefix = "TEMP";

package list_service;

service ListService {
  // Get list
  rpc GetList (EmptyMessage) returns (List) {}

  // Append value to list
  rpc Append (ListItem) returns (List) {}

  // Insert value to list
  rpc Insert (ListItemInsert) returns (List) {}

  // Delete value from list
  rpc Delete (ListItem) returns (List) {}

  // Find value in list
  rpc Find (ListItem) returns (ListItemPosition) {}

  // Sort list
  rpc Sort (ListSort) returns (List) {}
}

message EmptyMessage {}

message List {
  repeated int32 items = 1;
}

message ListItem {
  int32 item = 1;
}

message ListItemPosition {
  int32 position = 1;
}

message ListItemInsert {
  int32 item = 1;
  int32 position = 2;
}

message ListSort {
  bool reverse = 1;
}
